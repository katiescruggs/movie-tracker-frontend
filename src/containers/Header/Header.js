import React, { Component } from 'react';
import { Link, browserHistory } from 'react-router-dom';
import { connect } from 'react-redux';
import { fetchMovies, logout } from '../../actions';


class Header extends Component {

   async componentDidMount() {
     this.props.handleFetch();
   }

   componentWillUpdate(nextProps) {
    if(nextProps.signedIn && !this.props.signedIn) {
      this.props.history.push('/');
    }
   }

  render() {
    const name = this.props.name ? `${this.props.name}'s` : '';

    const favoritesButton = this.props.signedIn 
      ? <Link to="/favorites">
          <button
            className="btn-favorites"
          >
            Favorites
          </button>
        </Link>
      : null;

    const buttons = this.props.signedIn 
      ? <Link to='/'>
          <button 
            className='btn-log-in'
            onClick={this.props.handleLogout}
          >
            Log Out
          </button>
        </Link>

      : <div>
          <Link to='/login'>
            <button className='btn-log-in'>
              Log In
            </button>
          </Link>
          <Link to='/register'>
            <button className='btn-sign-up'>
              Sign Up
            </button>
          </Link>
        </div> 

    return (
      <header className="App-header">
        {favoritesButton}
        <h1 className="App-title">{`${name} MovieTracker`}</h1>
        {buttons}
      </header>
    );
  }
}

export const mapStateToProps = state => {
  return {
    name: state.user.info.name,
    signedIn: state.user.signedIn
  };
};

export const mapDispatchToProps = dispatch => {
  return {
    handleLogout: () => {
      dispatch(logout());
    },
    handleFetch: () => {
      dispatch(fetchMovies());
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);