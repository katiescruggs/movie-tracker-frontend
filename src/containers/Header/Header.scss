.App-header {
  background-color: #B22628;
  height: 100px;
  padding: 20px;
  text-align: center;
  & a {
    color: #fff;
  }
}

.btn-log-in {
  display: inline-block;
  position: absolute;
  top: 5px;
  right: 85px;
  background: none;
  border: none;
  color: #1c285b;
  font-size: 13pt;
  outline: none;
}

.btn-sign-up {
  position: absolute;
  top: 5px;
  right: 5px;
  background: none;
  border: none;
  font-size: 13pt;
  outline: none;
  color: #1c285b;
}

.btn-header-favorite {
  position: absolute;
  top: 5px;
  left: 5px;
  background: none;
  border: none;
  font-size: 13pt;
  outline: none;
  color: #1c285b;
}

.btn-sign-up,
.btn-log-in,
.btn-header-favorite {
  &:hover {
    cursor: pointer;
    color: white;
  }
}

.App-title {
  font-size: 50pt;
  font-family: 'CanterBold3D', sans-serif;
  display: inline-block;
  letter-spacing: .4em;
}
