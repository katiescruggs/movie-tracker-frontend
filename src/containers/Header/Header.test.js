import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { Header, mapStateToProps } from './Header';

describe('Header tests', () => {
  let header;
  beforeEach( () => {
    header = shallow(<Provider ><Header /></Provider>);
  });

  it('should exist', () => {
    expect(header).toBeDefined();
  });

  describe('mapStateToProps', () => {
    it('should pull user info from the store', () => {
      const mockStore = {
        name: 'Yung Jhun',
        signedIn: false
      };
      const result = mapStateToProps(mockStore);
      expect(result.name).toEqual(mockStore.name);
      expect(result.signedIn).toEqual(mockStore.signedIn);
    });
  });

  describe('mapDispatchToProps', () => {
    
  });
});