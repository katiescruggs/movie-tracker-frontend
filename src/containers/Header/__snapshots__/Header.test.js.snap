// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Container Header should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    handleFetch={[Function]}
    handleLogout={[Function]}
    name="Katie"
    signedIn={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/favorites"
>
          <button
                    className="btn-header-favorite"
          >
                    Favorites
          </button>
</Link>,
        <Link
          replace={false}
          to="/"
>
          <h1
                    className="App-title"
          >
                    MovieTracker
          </h1>
</Link>,
        <div
          className="welcome-msg"
>
          <p
                    className="btn-log-in"
          >
                    Welcome, Katie
          </p>
          <Link
                    replace={false}
                    to="/"
          >
                    <button
                              className="btn-sign-up"
                              onClick={[Function]}
                    >
                              Log Out
                    </button>
          </Link>
</div>,
      ],
      "className": "App-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="btn-header-favorite"
>
            Favorites
</button>,
          "replace": false,
          "to": "/favorites",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorites",
            "className": "btn-header-favorite",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h1
            className="App-title"
>
            MovieTracker
</h1>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "MovieTracker",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "MovieTracker",
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="btn-log-in"
>
              Welcome, Katie
</p>,
            <Link
              replace={false}
              to="/"
>
              <button
                            className="btn-sign-up"
                            onClick={[Function]}
              >
                            Log Out
              </button>
</Link>,
          ],
          "className": "welcome-msg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome, Katie",
              "className": "btn-log-in",
            },
            "ref": null,
            "rendered": "Welcome, Katie",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="btn-sign-up"
                onClick={[Function]}
>
                Log Out
</button>,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log Out",
                "className": "btn-sign-up",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Log Out",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/favorites"
>
            <button
                        className="btn-header-favorite"
            >
                        Favorites
            </button>
</Link>,
          <Link
            replace={false}
            to="/"
>
            <h1
                        className="App-title"
            >
                        MovieTracker
            </h1>
</Link>,
          <div
            className="welcome-msg"
>
            <p
                        className="btn-log-in"
            >
                        Welcome, Katie
            </p>
            <Link
                        replace={false}
                        to="/"
            >
                        <button
                                    className="btn-sign-up"
                                    onClick={[Function]}
                        >
                                    Log Out
                        </button>
            </Link>
</div>,
        ],
        "className": "App-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="btn-header-favorite"
>
              Favorites
</button>,
            "replace": false,
            "to": "/favorites",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorites",
              "className": "btn-header-favorite",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h1
              className="App-title"
>
              MovieTracker
</h1>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MovieTracker",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "MovieTracker",
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="btn-log-in"
>
                Welcome, Katie
</p>,
              <Link
                replace={false}
                to="/"
>
                <button
                                className="btn-sign-up"
                                onClick={[Function]}
                >
                                Log Out
                </button>
</Link>,
            ],
            "className": "welcome-msg",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome, Katie",
                "className": "btn-log-in",
              },
              "ref": null,
              "rendered": "Welcome, Katie",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn-sign-up"
                  onClick={[Function]}
>
                  Log Out
</button>,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Log Out",
                  "className": "btn-sign-up",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Log Out",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
